### PR TITLE
Fix error when importing .js libraries into nextjs app

### LINKED ISSUE #4904: Error importing javascript React component into NextJS app
## Current Behavior

I'm getting the following error importing javascript React components into an NextJS app:

```
error - /Users/sebas/Sites/test/libs/ui/src/lib/button/button.js 5:4
Module parse failed: Unexpected token (5:4)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export function Button(props) {
|   return (
>     <div>
|       <h1>Welcome to button!</h1>
|     </div>
```

This is not happening when the React components are typescript.

## Expected Behavior

Import React components without errors.

## Steps to Reproduce

```
$ npx create-nx-workspace --js

Workspace name: foo
What to create in the new workspace: react
Application name: bar
Default stylesheet format: SASS
Use Nx Cloud?: No

$ cd foo
$ nx g @nrwl/react:lib ui --js
$ nx g @nrwl/react:component button --project ui --js

Should this component be exported in the project? Y

$ npm install -D @nrwl/next
$ nx generate @nrwl/next:application --js

What name would you like to use for the application?: test
Which stylesheet format would you like to use?: SASS
```

Edit `apps/test/pages/index.js` and import the `Button` component:

```
import {Button} from '@foo/ui';
```

Serve the `test` app:

```
$ nx serve test
```

### Failure Logs

Log for the app `test`:

```
$ nx serve test

> nx run test:serve
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
info  - automatically enabled Fast Refresh for 1 custom loader
info  - Using external babel configuration from /Users/sebas/Sites/foo/apps/test/.babelrc
event - compiled successfully
[ ready ] on http://localhost:4200
event - build page: /next/dist/pages/_error
wait  - compiling...
event - compiled successfully
event - build page: /
wait  - compiling...
error - /Users/sebas/Sites/foo/libs/ui/src/lib/button/button.js 5:4
Module parse failed: Unexpected token (5:4)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export function Button(props) {
|   return (
>     <div>
|       <h1>Welcome to button!</h1>
|     </div>
event - compiled successfully
```

### Environment

```
$ nx report

>  NX  Report complete - copy this into the issue template

  Node : 14.15.5
  OS   : darwin x64
  npm  : 6.14.11

  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.4.0
  @nrwl/cypress : 11.4.0
  @nrwl/devkit : 11.4.0
  @nrwl/eslint-plugin-nx : 11.4.0
  @nrwl/express : Not Found
  @nrwl/jest : 11.4.0
  @nrwl/linter : 11.4.0
  @nrwl/nest : Not Found
  @nrwl/next : 11.4.0
  @nrwl/node : Not Found
  @nrwl/react : 11.4.0
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.4.0
  @nrwl/web : 11.4.0
  @nrwl/workspace : 11.4.0
  typescript : 4.0.7
```

### PR DESCRIPTION
**Pull request by sebacruz**
## Related Issue(s)

Fixes #4904

### CODE REVIEW

**Comment by kirjai** [comment]
Hey @sebacruz , the build didn't seem to pick up these changes, unfortunately.. that happens sometimes. can you try force pushing your branch, in an attempt for CircleCI to pick these changes up, please?

File: packages/next/src/utils/config.ts [hunk 1]
```diff
@@ -56,6 +56,11 @@ export function createWebpackConfig(
         test: /\.tsx?$/,
         use: [defaultLoaders.babel],
       },
+      {
+        test: /\.js?$/,
```
Comment (jaysoo) [inline packages/next/src/utils/config.ts:60]: Can we update the existing loader to test on `/\.(j|t)sx?$/` instead? This will capture `.ts,.tsx,.js,.jsx`

Comment (sebacruz) [reply packages/next/src/utils/config.ts]: Updated at f96d85f.

**Comment by kirjai** [comment]
Let me try pushing the branch, see if that solves it

**Comment by kirjai** [comment]
once again, thank you for the PR! 🎉

**Pull request merged**